apis/v1beta1/vspheremachine_types.go

```diff
@@ -42,6 +43,34 @@ type VSphereMachineSpec struct {
 	// FailureDomain is the failure domain unique identifier this Machine should be attached to, as defined in Cluster API.
 	// For this infrastructure provider, the name is equivalent to the name of the VSphereDeploymentZone.
 	FailureDomain *string `json:"failureDomain,omitempty"`
+
+	// PowerOffMode describes the desired behavior when powering off a VM.
+	//
+	// There are three, supported power off modes: hard, soft, and
+	// trySoft. The first mode, hard, is the equivalent of a physical
+	// system's power cord being ripped from the wall. The soft mode
+	// requires the VM's guest to have VM Tools installed and attempts to
+	// gracefully shut down the VM. Its variant, trySoft, first attempts
+	// a graceful shutdown, and if that fails or the VM is not in a powered off
+	// state after reaching the GuestSoftPowerOffTimeout, the VM is halted.
+	//
+	// If omitted, the mode defaults to trySoft.
+	//
+	// +optional
+	// +kubebuilder:default=trySoft
+	PowerOffMode vmopv1.VirtualMachinePowerOpMode `json:"powerOffMode,omitempty"`
+
+	// GuestSoftPowerOffTimeout sets the wait timeout for shutdown in the VM guest.
+	// The VM will be powered off forcibly after the timeout if the VM is still
+	// up and running when the PowerOffMode is set to trySoft.
+	//
+	// This parameter only applies when the PowerOffMode is set to trySoft.
+	// The reconciler will wait infinitely when it is set to zero.
+	//
+	// If omitted, the timeout defaults to 5 minutes.
+	//
+	// +optional
+	GuestSoftPowerOffTimeout *metav1.Duration `json:"guestSoftPowerOffTimeout,omitempty"`
```

Comment:
Should we add a validation webhook case that this is only allowed to get set if the Mode is set to `TrySoft`?

Another case: don't allow negative values or 0. (if 0 would be equal to `Soft`)